### PR TITLE
change up and add ninja objectives

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
instead of guaranteed hyper metagameable 3 sabotage objectives:
- fixed objectives are now just doorjack, bomb and staying alive
- you randomly get assigned 2 of 3 sabotage objectives
- you can get up to 2 killing gibbing or stealing objectives
  stealing is for sentimental things like lucky bill, ian photobook, caps cloak

## Why / Balance
less metagaming but also less friendly ninjas that dont do anything except "pls my sabotage isnt actually sabotage no arresti"
will be cool seeing ninjas kill a target then go invis and run around with their corpse for everyone to see :trollface:

## Technical details
yml gaming

moved TeachLessonCondition to BaseTeachLessonObjective because it didnt have it pro

## Media
some rolls:

## Requirements
- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
no

**Changelog**
:cl:
- tweak: Ninja objectives are now more random and they can be asked to steal things or kill people!

### CODE REVIEW

**Comment by Avalon-Proto** [comment]
Do the spider clan use cyborgs in other servers?

**Comment by Lyndomen** [comment]
Why teach lesson? Teach lesson is for kill once free greentext, not gib.

**Comment by Lyndomen** [comment]
> Do the spider clan use cyborgs in other servers?

No but cool flavor

**Comment by deltanedas** [comment]
> Why teach lesson? Teach lesson is for kill once free greentext, not gib.

yes thats intentional
you can rarely be told to gib a head (Execute X, Job)

**Comment by Lyndomen** [comment]
Ok I see, gib or teach lesson objs ty. Brought it up for discussion, I like the idea a lot, feedback pending

**Comment by deltanedas** [comment]
> Do the spider clan use cyborgs in other servers?

on ss13 you can convert borgs to be assault/saboteur/evil medi borgs as an objective, upstream didnt want it so i never added it when borgs got added

**Comment by Lyndomen** [comment]
Direction big 👍

**Review by Lyndomen — APPROVED** [review]
Actual approval ops

File: Resources/Locale/en-US/deltav/objectives/conditions/steal-target-groups.ftl [hunk 1]
```diff
@@ -6,3 +6,5 @@ steal-target-groups-notary-stamp = notary stamp
 steal-target-groups-silvia = silvia
 
 steal-target-groups-recruiter-pen = recruiter's pen
+
+steal-target-groups-captains-cloak = captain's cloak
```
Comment (MilonPL) [inline Resources/Locale/en-US/deltav/objectives/conditions/steal-target-groups.ftl]: what's up with these random spaces

File: Resources/Prototypes/DeltaV/Objectives/ninja.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# Kill
+
+- type: entity
+  abstract: true
+  parent: [BaseNinjaObjective, BaseTeachLessonObjective]
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Objectives/ninja.yml]: spaces in inline lists `[ Option1, Option2 ]`

File: Resources/Prototypes/DeltaV/Objectives/ninja.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  - type: PickRandomHead
+
+- type: entity
+  parent: BaseNinjaTeachLessonObjective
+  id: NinjaTeachLessonRandomPersonObjective
+  components:
+  - type: PickRandomPerson
+
+- type: entity
+  parent: [BaseNinjaObjective, BaseKillObjective]
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Objectives/ninja.yml]: .

**Review by MilonPL — CHANGES_REQUESTED** [review]

**Review by MilonPL — APPROVED** [review]

**Pull request merged**